pkg/apiserver/slowquery/statement_gen.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slowquery
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/thoas/go-funk"
+)
+
+func (s *Service) genSelectStmt(tableColumns []string, reqFields []string) string {
+	fields := getFieldsAndTags()
+
+	// use reqFields filter when not all fields are requested
+	if reqFields[0] != "*" {
+		// These three fields are the most basic information of a slow query record and should contain them
+		reqFields = funk.UniqString(append(reqFields, "digest", "connection_id", "timestamp"))
+		fields = funk.Filter(fields, func(f Field) bool {
+			return funk.Contains(reqFields, f.JSON)
+		}).([]Field)
+	}
+
+	// filter by tableColumns
```

Comment:
// We have both TiDB 4.x and TiDB 5.x columns listed in the model. Filter out columns that do not exist in current version TiDB schema.